ps_democqrshooksusage.php

```diff
@@ -0,0 +1,337 @@
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+
+//todo: demonstrate how include custom js extensions for existing grids maybe?.
+//todo: not a single translation works for this module
+/**
+ * Class Ps_DemoCQRSHooksUsage demonstrates the usage of CQRS and hooks.
+ */
+class Ps_DemoCQRSHooksUsage extends Module
+{
+    public function __construct()
+    {
+        $this->name = 'ps_democqrshooksusage';
+        $this->version = '1.0.0';
+        $this->author = 'Tomas Ilginis';
+        $this->need_instance = 0;
+
+        parent::__construct();
+
+        $this->displayName = $this->getTranslator()->trans(
+            'Demo for CQRS and hooks usage',
+            [],
+            'Modules.Ps_DemoCQRSHooksUsage'
+        );
+
+        $this->description =
+            $this->getTranslator()->trans(
+                'Help developers to understand how to create module using new hooks and apply best practices when using CQRS',
+                [],
+                'Modules.Ps_DemoCQRSHooksUsage'
+            );
+
+        $this->ps_versions_compliancy = [
+            'min' => '1.7.6.0',
+            'max' => _PS_VERSION_,
+        ];
+    }
+
+    /**
+     * Install module and register hooks to allow grid modification.
+     *
+     * @return bool
+     */
+    public function install()
+    {
+        return parent::install() &&
+            // Register hook to allow Customer grid definition modifications.
+            // Each grid's definition modification hook has it's own name. Hook name is built using
+            // this structure: "action{grid_id}GridDefinitionModifier", in this case "grid_id" is "customer"
+            // this means we will be modifying "Sell > Customers" page grid.
+            // You can check any definition factory service in PrestaShop\PrestaShop\Core\Grid\Definition\Factory
+            // to see available grid ids. Grid id is returned by `getId()` method.
+            $this->registerHook('actionCustomerGridDefinitionModifier') &&
+            // Register hook to allow Customer grid query modifications which allows to add any sql condition.
+            $this->registerHook('actionCustomerGridQueryBuilderModifier') &&
+            // Register hook to allow overriding customer form
+            // this structure: "action{block_prefix}FormBuilderModifier", in this case "block_prefix" is "customer"
+            // {block_prefix} is either retrieved automatically by its type. E.g "ManufacturerType" will be "manufacturer"
+            // or it can be modified in form type by overriding "getBlockPrefix" function
+            $this->registerHook('actioncustomerFormBuilderModifier') &&
+            $this->registerHook('actionAfterCreatecustomerFormHandler') &&
```

Comment:
```suggestion
            $this->registerHook('actionAfterCreateCustomerFormHandler') &&
```
Let's check it's really a minor c, because in the hook list, it's a upper C
https://github.com/PrestaShop/PrestaShop/blob/develop/install-dev/data/xml/hook.xml#L966